File: camerax_android/lib/src/core/zoom_state_impl.dart
import 'package:camerax_android/src/api.dart';
import 'package:camerax_platform_interface/camerax_platform_interface.dart';

final class ZoomStateImpl implements ZoomState {
  final ZoomStateProxyApi api;

  ZoomStateImpl.internal(this.api);

  @override
  double get linearZoom => api.linearZoom;
  @override
  double get maxZoomRatio => api.maxZoomRatio;
  @override
  double get minZoomRatio => api.minZoomRatio;
  @override
  double get zoomRatio => api.zoomRatio;
}

extension ZoomStateProxyApiX on ZoomStateProxyApi {
  ZoomState get impl => ZoomStateImpl.internal(this);
}

File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/core/ZoomStateImpl.kt
package dev.hebei.camerax_android.core

import dev.hebei.camerax_android.CameraXImpl
import dev.hebei.camerax_android.PigeonApiZoomStateApi

class ZoomStateImpl(impl: CameraXImpl) : PigeonApiZoomStateApi(impl) {
    override fun minZoomRatio(pigeon_instance: androidx.camera.core.ZoomState): Double {
        return pigeon_instance.minZoomRatio.toDouble()
    }

    override fun maxZoomRatio(pigeon_instance: androidx.camera.core.ZoomState): Double {
        return pigeon_instance.maxZoomRatio.toDouble()
    }

    override fun zoomRatio(pigeon_instance: androidx.camera.core.ZoomState): Double {
        return pigeon_instance.zoomRatio.toDouble()
    }

    override fun linearZoom(pigeon_instance: androidx.camera.core.ZoomState): Double {
        return pigeon_instance.linearZoom.toDouble()
    }
}